### PR TITLE
626 interval selections

### PR DESCRIPTION
**Pull request by martehvi**

### CODE REVIEW

File: packages/frontend/src/components/SimpleDatePicker.vue [hunk 1]
```diff
@@ -38,7 +44,7 @@ export default Vue.extend({
       if (this.currentDate === target.value) {
         return;
       }
-
+      console.log("the gold");
```
Comment (klyngen) [inline packages/frontend/src/components/SimpleDatePicker.vue:47]: Remove console logs

File: packages/frontend/src/components/InvoiceStatistics.vue [hunk 2]
```diff
@@ -93,7 +106,7 @@ export default Vue.extend({
   gap: 0.5rem;
 }
 
-.statistic-cards > div {
+.statistic-cards>div {
```
Comment (klyngen) [inline packages/frontend/src/components/InvoiceStatistics.vue:109]: Nit: add spacing

File: packages/frontend/src/components/IntervalShortcuts.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<template>
+  <md-button class="md-primary md-raised" @click="onIntervalClicked()">{{ typeInterval }}</md-button>
+</template>
+  
+<script lang="ts">
+import Vue from "vue";
+import moment, { Moment } from "moment";
+export default Vue.extend({
+  props: {
+    defaultDate: {
```
Comment (klyngen) [inline packages/frontend/src/components/IntervalShortcuts.vue:10]: Don't think this prop is used anywhere

File: packages/frontend/src/components/IntervalShortcuts.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      currentDate: this.defaultDate || "",
+      fromDate: this.defaultDate || "",
+      toDate: this.defaultDate || ""
+    };
+  },
+  methods: {
+    onIntervalClicked() {
+
+      var today = new Date();
+      var month = today.getMonth() + 1;
```
Comment (klyngen) [inline packages/frontend/src/components/IntervalShortcuts.vue:34]: There is a lot of repetitive code in this function and I know we can do better.

How I would think about a component like this:
- property label: string
- property months: number

Emits a toDate and a fromDate

No need to have a switch that switches on strings 😄

And the variables in data can be removed when those changes are done. 

I can explain in more detail if you have questions.

File: packages/frontend/src/components/InvoiceStatisticCard.vue [hunk 5]
```diff
@@ -3,9 +3,9 @@
     <h3 class="card-title">{{ cardData.title }}</h3>
     <div class="section" v-for="value in cardData.values" :key="value.title">
       <h4>{{ value.title }}</h4>
-      <div class="value">
-        <h4>{{ value.value }}</h4>
-        <p>{{ value.unit }}</p>
+      <div class="content">
+        <div class="value"><b>{{ value.value }}</b></div>
```
Comment (klyngen) [inline packages/frontend/src/components/InvoiceStatisticCard.vue:7]: Semantics save lives 🎉

**Review by klyngen — CHANGES_REQUESTED** [review]
A few comments to lift the quality 🏋️

File: packages/frontend/src/components/IntervalShortcuts.vue [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      var year = today.getFullYear();
+
+      switch (this.typeInterval) {
+        case "Siste halvår":
+          var lasthalfyear = new Date();
+          lasthalfyear.setMonth(month - 6);
+          this.toDate = moment(today).format("YYYY-MM-DD");
```
Comment (klyngen) [inline packages/frontend/src/components/IntervalShortcuts.vue:41]: Don't think the usage of moment is needed. There are some methods in the invoice-store. Extract those methods into a util file and use the same component here. All dependencies in Javascript are technical debt 🤓

**Review by klyngen — CHANGES_REQUESTED** [review]

**Comment by Fordiboii** [comment]
Closing, implemented in https://github.com/Alv-no/alvtime/pull/690

**Pull request closed without merging**